### ISSUE TITLE
Create `intToBytes` function for Python 3 compatibility with Python 2

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6053]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#6053|
|Type|enhancement|
|Created|2012-10-04 14:50:08Z|
|Branch|https://github.com/twisted/twisted/tree/toBytes-py3-6053|

On Python 2, it's possible to do things like:

```
>>> bytes(20)
'20'
```

On Python 3 that will not work. Since we use the above idiom quite a bit, at the very least in tests, we should implement a `intToBytes` function that emulates this behavior.

<details><summary>Searchable metadata</summary>

```
trac-id__6053 6053
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_toBytes_py3_6053 branches/toBytes-py3-6053
branch_author__itamarst itamarst
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1349362208000000 1349362208000000
changetime__1349463476000000 1349463476000000
version__None None
owner__itamar itamar

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_1"></a>|@itamarst set owner to @exarkun|
|-|-|

http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/toBytes-py3-6053

I couldn't think of a better way to the full compatibility, unfortunately. We're going to need this if we ever do `%s` support, but it's easy enough to drop as YAGNI. If we do drop it, we should probably make this API private, since it's easy to get it wrong if you use it for anything other than integers.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_2"></a>|@itamarst commented|
|-|-|

(In [35943]) Branching to 'toBytes-py3-6053'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @itamarst|
|-|-|

  1. Using `memoryview` for this seems like a bad idea.  Exactly three types are supported (exactly three types are unit tested).  Wandering into the "buffer API" realm is just asking for trouble.  All we really need is a function for getting a bytes-representation of a few very simple types.  Who knows what dragging `memoryview` into the picture is going to do?  Let's implement and advertise this as a very simple function of limited capabilities.  If it really turns out to be necessary, we can expand that simple functionality later.
  1. `intToBytes`, `listToBytes` might be an alternative approach that suggests different (_slightly_ better) names.
  1. `__all__` needs to be updated.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst commented|
|-|-|

(In [35971]) Replace toBytes with intToBytes. Refs [#6053](https://github.com/twisted/twisted/issues/6053)

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_5"></a>|@itamarst set owner to @exarkun|
|-|-|

I've decided to just make `intToBytes`.

http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/toBytes-py3-6053 is a new run.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set owner to @itamarst|
|-|-|

Looks good, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_7"></a>|@itamarst set status to `closed`|
|-|-|

(In [35984]) Merge toBytes-py3-6053.

Author: itamar
Review: exarkun
Fixes: [#6053](https://github.com/twisted/twisted/issues/6053)

intToBytes utility function.
